File: src/Aardvark.OpenCV.Tests/ClusteringTests.cs
using Aardvark.Base;
using NUnit.Framework;

namespace Aardvark.OpenCV.Tests
{
    [TestFixture]
    internal class ClusteringTests
    {
        private static Vector<float> Vec(float v) => new([v, v, v]);

        [OneTimeSetUp]
        public void Init()
        {
            Aardvark.Base.Aardvark.Init();
        }

        [Test]
        public void KMeans()
        {
            Vector<float>[] data = { Vec(-1.2f), Vec(-1.3f), Vec(-1.4f), Vec(2.3f), Vec(2.4f) };
            OpenCVKMeansClustering.ClusterKMeans(data, 2, 1, false, out var clusters, out var centers);
            Assert.AreEqual(new int[] { 0, 0, 0, 1, 1 }, clusters);
        }
    }
}

File: src/Aardvark.OpenCV.Tests/Tests/ClusteringTests.fs
namespace Aardvark.OpenCV.Tests

open Aardvark.Base
open Aardvark.OpenCV
open NUnit.Framework
open FsUnit

module ``Clustering Tests`` =

    let private vec value =
        Vector<float32> [| value; value; value |]

    [<SetUp>]
    let init() =
        Aardvark.Init()

    [<Test>]
    let ``K-means``() =
        let data = [| vec -1.2f; vec -1.3f; vec -1.4f; vec 2.3f; vec 2.4f |]

        let mutable clusters = Array.empty
        let mutable centers = Array.empty
        OpenCVKMeansClustering.ClusterKMeans(data, 2, 1, false, &clusters, &centers);

        clusters |> should equal [| 0; 0; 0; 1; 1 |]

